### ISSUE TITLE
[Bug]: native unit %RH is not expected in device class

### ISSUE THREAD

**Issue by criticallimit** [custom-components/ble_monitor#1167]
### What happened?

```
Logger: homeassistant.components.sensor
Source: components/sensor/__init__.py:648 
Integration: Sensor (documentation, issues) 
First occurred: 17:12:18 (1 occurrences) 
Last logged: 17:12:18

Entity sensor.ble_cypress_humidity_1ca92e23f0874df7b9a2fd4b716a4bf6 (<class 'custom_components.ble_monitor.sensor.HumiditySensor'>) is using native unit of measurement 'RH%' which is not a valid unit for the device class ('humidity') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
```

### Sensor type

Temperture and humidity sensor

**Comment by criticallimit** [comment]
```
Logger: homeassistant.components.sensor
Source: components/sensor/__init__.py:648 
Integration: Sensor (documentation, issues) 
First occurred: 17:12:18 (1 occurrences) 
Last logged: 17:12:18

Entity sensor.ble_cypress_humidity_1ca92e23f0874df7b9a2fd4b716a4bf6 (<class 'custom_components.ble_monitor.sensor.HumiditySensor'>) is using native unit of measurement 'RH%' which is not a valid unit for the device class ('humidity') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
```

**Comment by Ernst79** [comment]
Thanks for reporting. Should be fixed in 12.0.0-beta
